pkg/apicheck/check.go

```diff
@@ -130,14 +130,23 @@ func (c *ApiConnectivityCheck) getWorkerPeersResponse() peers.Response {
 
 	c.config.Log.Info("Error count exceeds threshold, trying to ask other nodes if I'm healthy")
 	peersToAsk := c.config.Peers.GetPeersAddresses(peers.Worker)
-	if peersToAsk == nil && c.config.MinPeersForRemediation != 0 || len(peersToAsk) < c.config.MinPeersForRemediation {
-		c.config.Log.Info("Peers list is empty and / or less than the minimum required peers for remediation, so consider the node being healthy")
-		//todo maybe we need to check if this happens too much and reboot
+
+	// We check to see if we have at least the number of peers that the user has configured as required.
+	//	If we don't have this many peers (for instance there are zero peers, and the default value is set
+	//	which requires at least one peer), we don't want to remediate. In this case we have some confusion
+	//	and don't want to remediate a node when we shouldn't.  Note: It would be unusual for MinPeersForRemediation
+	//	to be greater than 1 unless the environment has specific requirements.
+	if len(peersToAsk) < c.config.MinPeersForRemediation {
+		c.config.Log.Info("Peers list is empty and / or less than the minimum required peers for remediation, " +
```

Comment:
Log message can be confusing for this use case.
Assuming we don't use only 1 or 0 values for `MinPeersForRemediation ` Peer List might not be empty.